src/Autofac/Core/Registration/DefaultRegisteredServicesTracker.cs

```diff
@@ -46,6 +51,12 @@ internal class DefaultRegisteredServicesTracker : Disposable, IRegisteredService
         /// </summary>
         private readonly object _synchRoot = new object();
 
+        public DefaultRegisteredServicesTracker()
```

Comment:
Certainly seems to be (that's been there a while); I find myself wondering why the events are stored in this properties object rather than just as regular events? Any idea? Nothing sets those properties outside the class, and there is a separate properties instance in `ComponentRegistration` that contains the actual registry properties.

Reply:
I honestly don't know. Could be a remnant of refactoring that didn't get cleaned up. I just happened to be stepping through some of the flows to get a better understanding and saw this, was like... uh... hmmm. Looks like it might have been part of #1077 which was to ensure component registration events continued to fire correctly.

I feel like it could probably just be a standard event handling mechanism, though. I'm not sure what benefit the dictionary approach provides here.

Reply:
I've removed the `_properties` dictionary, just using regular events now.  No obvious consequences (apart from using less memory).